File: JS/Basics/4_5_armstrong.js
// a number equal to sum of its digits raised to no of digits

function armstrong(n) {
    let nStr = String(n);
    let len = nStr.length;

    let sum = 0;
    let num = n;
    while (num > 0) {
        let rem = num % 10;
        sum += Math.pow(rem, len);

        num = Math.floor(num / 10);
    }

    if (sum == n) console.log("Armstrong number.");
    else console.log("Not an armstrong number");
}

armstrong(155);

File: C/Basics/4_5_armstrong.c
/*
    identify if the number is armstrong number
*/

#include <stdio.h>
#include <string.h>
#include <math.h>

int armstrongNumber(int N)
{
    int num = N, sum = 0, rem, digits = 0;

    // find size of digits
    char numInStr[10];
    sprintf(numInStr, "%d", N);
    digits = strlen(numInStr);

    // Add sum of powerss
    while (num != 0)
    {
        rem = num % 10;
        num = num / 10;
        sum += round(pow(rem, digits)); // pow  returns power in floting point numbers so we need to round it
        // eg ... it returns 124 for 5^3 after rounding it becomes 125
    }

    printf("Sum : %d\n", sum);
    return sum == N;
}

int main()
{

    int N = 153;
    printf("Enter a number : ");
    scanf("%d", &N);

    if (armstrongNumber(N))
        printf("Entered number : %d is an armstrong number !\n ", N);
    else
        printf("Entered number : %d is not an armstrong number !\n ", N);
}